src/Raven.Server/Commercial/SetupInfo.cs

```diff
@@ -0,0 +1,160 @@
+            {
+                return new DynamicJsonValue
+                {
+                    [nameof(NodeTag)] = NodeTag,
+                    [nameof(ServerUrl)] = ServerUrl,
+                    [nameof(PublicServerUrl)] = PublicServerUrl,
+                    [nameof(Port)] = Port,
+                    [nameof(Hostname)] = Hostname,
+                    [nameof(Certificate)] = Certificate,
+                    [nameof(Ips)] = Ips.ToArray(),
+                };
+            }
+        }
+    }
+
+    public class UnsecuredSetupInfo
+    {
+        public string ServerUrl { get; set; }
+        public string PublicServerUrl { get; set; }
+
+        public DynamicJsonValue ToJson()
+        {
+            return new DynamicJsonValue
+            {
+                [nameof(ServerUrl)] = ServerUrl,
+                [nameof(PublicServerUrl)] = PublicServerUrl
+            };
+        }
+    }
+    
+    public class ClaimDomainInfo
+    {
+        public License License { get; set; }
+        public string Domain { get; set; }
+
+        public DynamicJsonValue ToJson()
+        {
+            return new DynamicJsonValue
+            {
+                [nameof(License)] = License.ToJson(),
+                [nameof(Domain)] = Domain
+            };
+        }
+    }
+
+    public class RegistrationResult
+    {
+        public RegistrationStatus Status { get; set; }
+
+        public DynamicJsonValue ToJson()
+        {
+            return new DynamicJsonValue
+            {
+                [nameof(Status)] = Status.ToString()
+            };
+        }
+    }
+
+    public enum RegistrationStatus
+    {
+        Pending,
+        Done,
+        Error
+    }
+
+    public enum SetupMode
+    {
+        Initial,
+        LetsEncrypt,
+        Secured,
+        Unsecured
+    }
+
+    public class SetupProgressAndResult : IOperationResult, IOperationProgress
+    {
+        public long Processed { get; set; }
+        public long Total { get; set; }
+        public readonly List<string> Messages;
+
+        public SetupProgressAndResult()
+        {
+            Messages = new List<string>();
+        }
+
+        public string Message { get; private set; }
+
+        public DynamicJsonValue ToJson()
+        {
+            return new DynamicJsonValue(GetType())
+            {
+                [nameof(Processed)] = Processed,
+                [nameof(Total)] = Total,
+                [nameof(Messages)] = Messages
+        };
+        }
+
+        public void AddWarning(string message)
+        {
+            AddMessage("WARNING", message);
+        }
+
+        public void AddInfo(string message)
+        {
+            AddMessage("INFO", message);
+        }
+
+        public void AddError(string message)
+        {
+            AddMessage("ERROR", message);
+        }
+
+        private void AddMessage(string type, string message)
+        {
+            Message = $"[{SystemTime.UtcNow:T} {type}] {message}";
```

Comment:
Remember last message here, if identical to the previous one, make sure that we remove the previous one and insert this one instead. So this will just update the timestamp